src/main/java/games/strategy/triplea/ai/proAI/ProPurchaseAI.java

```diff
@@ -690,11 +786,11 @@ private void purchaseLandUnits(final Map<Territory, ProPurchaseTerritory> purcha
 
       // Determine most cost efficient units that can be produced in this territory
       final List<ProPurchaseOption> landFodderOptions =
-          ProPurchaseUtils.findPurchaseOptionsForTerritory(player, purchaseOptions.getLandFodderOptions(), t);
+          ProPurchaseUtils.findPurchaseOptionsForTerritory(player, purchaseOptions.getLandFodderOptions(), t, isBid);
       final List<ProPurchaseOption> landAttackOptions =
-          ProPurchaseUtils.findPurchaseOptionsForTerritory(player, purchaseOptions.getLandAttackOptions(), t);
+          ProPurchaseUtils.findPurchaseOptionsForTerritory(player, purchaseOptions.getLandAttackOptions(), t, isBid);
       final List<ProPurchaseOption> landDefenseOptions =
-          ProPurchaseUtils.findPurchaseOptionsForTerritory(player, purchaseOptions.getLandDefenseOptions(), t);
+          ProPurchaseUtils.findPurchaseOptionsForTerritory(player, purchaseOptions.getLandDefenseOptions(), t, isBid);
```

Comment:
Not really following your train of thought here. Each call is passing a different list of purchase options so yes it does need called 3 separate times. Think of it like this, you have 3 lists (A, B, C) and need to filter each of them. So its essentially calling:
```
filter(A)
filter(B)
filter(C)
```